vowpalwabbit/array_parameters.h

```diff
@@ -0,0 +1,172 @@
+#pragma once
+#include <stdio.h>
+#include <iostream>
+#include <stdlib.h>
+#include <string.h>
+
+#ifndef _WIN32
+#include <sys/mman.h>
+#endif
+
+using namespace std;
+typedef float weight;
+
+class weight_vector;
+
+class weights_iterator_iterator
+{
+private:
+	weight* _cur;
+public:
+	weights_iterator_iterator(weight* cur)
+		: _cur(cur)
+	{ }
+	
+	weight& operator*() { return *_cur; }
+
+	weights_iterator_iterator& operator++()
+	{
+		_cur++;
+		return *this;
+	}
+
+	template<typename T>
+	weights_iterator_iterator operator+(T index) { return weights_iterator_iterator(_cur + index); }
+
+	template<typename T>
+	weights_iterator_iterator& operator+=(T index)
+	{
+		_cur += index;
+		return *this;
+	}
+
+	weights_iterator_iterator& operator=(const weights_iterator_iterator& other)
```

Comment:
Why did you define an operator=, but not a copy or move constructor?  For one, the assignment operator is not necessary, as it will do the same thing as the default generated one.
Also, in general, C++ has the rule of 3/5 - if you write a copy/move/assignment/move assignment/destructor, you should consider the writing the other functions, too!